schemas/skills/samples/complex-skillmanifest-2.1.preview-0.json

```diff
@@ -0,0 +1,184 @@
+    "version": "1.0",
+    "description": "This is a sample skill definition for multiple activity types",
+    "publisherName": "Microsoft",
+    "privacyUrl": "https://myskill.contoso.com/privacy.html",
+    "copyright": "Copyright (c) Microsoft Corporation. All rights reserved.",
+    "license": "",
+    "iconUrl": "https://myskill.contoso.com/icon.png",
+    "tags": [
+        "sample",
+        "travel",
+        "weather"
+    ],
+    "endpoints": [
+        {
+            "name": "americas",
+            "protocol": "BotFrameworkV3",
+            "description": "Production  endpoint for SkillBot in the Americas",
+            "endpointUrl": "http://myskill.contoso.com/api/messages",
+            "msAppId": "00000000-0000-0000-0000-000000000000"
+        },
+        {
+            "name": "eu",
+            "protocol": "BotFrameworkV3",
+            "description": "Production  endpoint for SkillBot in Europe",
+            "endpointUrl": "http://myskill.contoso.com/api/messages",
+            "msAppId": "11111111-0000-0000-0000-000000000000"
+        }
+    ],
+    "dispatchModels": {
+        "languages": {
+            "en": [
+                {
+                    "id": "SkillLuModel-en",
+                    "name": "SkillBot LU (English)",
+                    "contentType": "application/lu",
+                    "url": "http://sample.com/SkillBot-en.lu",
+                    "description": "English language model for the skill"
+                },
+                {
+                    "id": "SkillQnAModel-en",
+                    "name": "SkillBot QnA LU (English)",
+                    "contentType": "application/lu",
+                    "url": "http://sample.com/SkillBot-QnA-en.lu",
+                    "description": "English language model for the skill (QnAMaker"
+                }
+            ],
+            "es-ES": [
+                {
+                    "id": "SkillLuModel-es-ES",
+                    "name": "SkillBot LU (Spanish-Spain)",
+                    "contentType": "application/lu",
+                    "url": "http://sample.com/SkillBot-es-ES.lu",
+                    "description": "Spanish (Spain) language model for the skill"
+                }
+            ],
+            "es-MX": [
+                {
+                    "id": "SkillLuModel-es-MX",
+                    "name": "SkillBot LU (Spanish-Mexico)",
+                    "contentType": "application/lu",
+                    "url": "http://sample.com/SkillBot-es-MX.lu",
+                    "description": "Spanish (Mexico) language model for the skill"
+                }
+            ]
+        },
+        "intents": {
+            "bookFlight": "#/activities/bookFlight",
+            "getWeather": "#/activities/getWeather",
+            "*": "#/activities/message"
+        }
+    },
+    "activities": {
+        "bookFlight": {
+            "description": "Books a flight",
+            "type": "event",
```

Comment:
What is the different between an event or an invoke (below?) beyond the activitytype ;-). Are we recommending one over the other?

Reply:
An event uses http back and forth (like we normally use). 

An invoke is request/reply, sort of a function call, it is always single turn and the caller will get some return value in the response object.

There is a sample invoke in the [DialogToDialog](https://github.com/microsoft/BotBuilder-Samples/blob/master/experimental/skills/DialogToDialog/DialogRootBot/Dialogs/MainDialog.cs#L139)  sample

Reply:
Great - so invokes would not return any activities / messages during processing? If so makes sense. Do you know if PVA plan to support invokes?